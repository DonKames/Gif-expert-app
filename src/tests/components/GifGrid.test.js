import '@testing-library/jest-dom';

import { shallow } from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Pruebas <GifGrid />', () => {
    const category = 'One Punch';

    useFetchGifs.mockReturnValue({
        data: [],
        loading: true
    });

    let wrapper = shallow(<GifGrid category= {category}/>);

    test('Debe Coincidir con Snapshot', () => {
        
        expect(wrapper).toMatchSnapshot();

    })
    
    test('Debe mostrar items cuando se cargan imagenes "useFetchGifs"', () => {
        const gifs = [{
            id: 'ABC',
            url: 'http://localhost/asdfasdf/img.jpg',
            title: 'La imagen'
        },
        {
            id: 'A23',
            url: 'http://localhost/asdfasdf/img23.jpg',
            title: 'La imagen 23'
        }]
        
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });
        let wrapper = shallow(<GifGrid category= {category}/>);

        expect(wrapper).toMatchSnapshot();

        expect(wrapper.find('p').exists()).toBe(false);

        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
    })
    
})
