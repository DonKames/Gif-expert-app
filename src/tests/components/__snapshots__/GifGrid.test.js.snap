// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas <GifGrid /> Debe Coincidir con Snapshot 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    One Punch
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Pruebas <GifGrid /> Debe mostrar items cuando se cargan imagenes "useFetchGifs" 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    One Punch
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="La imagen"
      url="http://localhost/asdfasdf/img.jpg"
    />
    <GifGridItem
      id="A23"
      key="A23"
      title="La imagen 23"
      url="http://localhost/asdfasdf/img23.jpg"
    />
  </div>
</Fragment>
`;
